book/HAS/hybrid-application-service-design.md

```diff
@@ -0,0 +1,88 @@
+
+### Importing from an Image
+HAC will prompt the user to enter an image to import a Component from. The Component CR controller will create a basic devfile with the image specified and store in the status field
+
+### Component Creation Flow
+After all of the individual components have been selected by the user, and the Application CR has been created, HAC will create Component CRs for each individual component. The HAS Component controller will:
+1. Validate the required fields in the Component CR
+   1. Does the Application CR exist
+   2. Does it point to a valid Sample/Github repo/devfile?
+2. Retrieve the devfile from the specified source, and validate it
+3. Add the devfile to the Component CR’s status
+4. Update the Application CR with an entry to the newly created Component
+
+## Application CR Update
+
+Application model update can be done by updating an existing Application CR directly. Application CR update is handled by the Application Controller. Only the mutable fields listed in the required fields in the Application CR can be updated.
+
+When an Application CR is being updated, the Application Controller will:
+1. Check to make sure all changed fields are mutable. Any changes on the immutable fields will result in an error in the Application CR Update.
+2. Update the application devfile (as specified in the spec.appModelRepository.url if applicable)
+
+## Component CR Update
+
+Component model update can be done by updating an existing Component CR directly. Component CR update is handled by the Component Controller. Only the mutable fields listed in the required fields in the Application CR can be updated.
+
+When an Component CR is being updated, the Component Controller will:
+1. Check to make sure all changed fields are mutable. Any changes on the immutable fields will result in an error in the Component CR Update.
+2. Update the component devfile (as specified in the status.data if applicable)
+3. For changes that require GitOps repo changes, call the HAS internal Push function to update the GitOps repository to reflect the model change for fields.
+
+## Usage of Devfile in Applications and Components
+
+The git source repo may or may not have the devfile physically stored within the repo. In case the repo has a devfile exists, it may not be the latest version. Future plan is to provide a way for creating PRs to put the latest devfile in the user’s source repo, the user may not accept the PR. Therefore, the devfile within the Application CRs and Component CRs will always be the latest and any action that needs to use the devfile stored in the Component CR to make sure it is using the latest version of the devfile, e.g. build pipeline and GitOps resource generation.
```

Comment:
I'm not sure if this is describing a limitation and future support.  I would remove any statements of future plans and just explain what support we have today

Reply:
Agreed, updated to what we have today.